components/identity.oauth.uma.permission.service/src/main/java/org/wso2/carbon/identity/oauth/uma/service/ReadPropertiesFile.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.oauth.uma.service;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.oauth.uma.service.model.PermissionTicketDO;
+import org.wso2.carbon.utils.CarbonUtils;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Paths;
+import java.util.Properties;
+
+/**
+ * ReadPropertiesFile is used to read configuration values from a properties file which is
+ *required for a permission ticket generation.
+ */
+//TODO: Move this configuration to identity.xml
+public class ReadPropertiesFile {
+
+    private static Log log = LogFactory.getLog(ReadPropertiesFile.class);
+
+    /**
+     * @param permissionTicketDO Configuration values for permission ticket
+     */
+    public static void readFileConfigValues(PermissionTicketDO permissionTicketDO) {
```

Comment:
To me this is bit confusing. What if reading this config fails ? DO will not have configs. Doesn't it harmful for the rest of the flow ? Can you continue without those configs ? Don't you need to break the flow ? Also setting these values to DO within this method doesn't sound right. May be you can return the configs from this method and set them to DO at a different level

Reply:
Also does this file reading happen per request ? If so we need to avoid this

Reply:
yes this is happening per request. It was suggested to later move this configuration to identity.xml